spec/game_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'spec_helper'
+require_relative '../dependency'
+
+RSpec.describe Game do
+  subject(:game) { described_class.new(name: 'Rspec', difficulty: 'Easy') }
+
+  describe '.check' do
+    it 'returns ++++' do
+      expect(game.check(game.secret)).to eq('++++')
+    end
+  end
+
+  describe '#use_hint' do
+    it 'decrements hints and calls a hint method if hints > 0' do
+      game.instance_variable_set(:@hints, 1)
+      game.use_hint
+      expect(game.hints).to eq(0)
```

Comment:
You can use just `change` matcher
https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/built-in-matchers/change-matcher